lib/cmd_grepip.go

```diff
@@ -121,10 +145,46 @@ func CmdGrepIP(
 
 	// prepare regexp
 	var rexp *regexp.Regexp
-	if ipv == 4 {
+	if ipv == 4 && f.CIDRsOnly && f.RangesOnly {
```

Comment:
Can we find a grouping that makes this more readable? If nesting `if`s helps then let's try that. Needs a judgement